app/models/event.rb

```diff
@@ -18,4 +19,17 @@ def data
       status: status
     }
   end
+
+  private
+
+  def date_validation
+    return unless !self[:finish_date].nil? && !self[:start_date].nil?
```

Comment:
Do not use `self`.
https://chodounsky.net/2014/05/07/when-to-use-self-in-activerecord-models/
`finish_date.present?`